File: reorder_menu.js

const fs = require('fs');
const path = 'd:/SOFT-MEDIC/Antigravity/CURARE/frontend/src/components/Layout.tsx';

try {
    const data = fs.readFileSync(path, 'utf8');
    // split by carriage return+newline or just newline to handle CRLF/LF
    const lines = data.split(/\r?\n/);

    console.log(`Total original lines: ${lines.length}`);

    // Ranges derived from original file analysis (0-indexed)
    // [start, end)
    const header = lines.slice(0, 109);
    const inicio = lines.slice(109, 124);
    const correos = lines.slice(124, 137);
    const agenda = lines.slice(137, 154);
    const usuarios = lines.slice(154, 171);
    const doctores = lines.slice(171, 234);
    const pacientes = lines.slice(234, 332);
    const personal = lines.slice(332, 400);
    const proveedores = lines.slice(400, 482);
    const arancel = lines.slice(482, 497);
    const egresos = lines.slice(497, 511);
    const laboratorios = lines.slice(511, 623);
    const gastos = lines.slice(623, 638);
    const inventario = lines.slice(638, 654);
    const hoja_diaria = lines.slice(654, 670);
    const utilidades = lines.slice(670, 684);
    const estadisticas = lines.slice(684, 787);
    const configuracion = lines.slice(787, 918);
    const footer = lines.slice(918);

    // New Order
    const newOrder = [
        ...header,
        ...inicio,
        ...agenda,
        ...pacientes,
        ...doctores,
        ...laboratorios,
        ...proveedores,
        ...arancel,
        ...personal,
        ...inventario,
        ...egresos,
        ...gastos,
        ...hoja_diaria,
        ...utilidades,
        ...estadisticas,
        ...correos,
        ...usuarios,
        ...configuracion,
        ...footer
    ];

    console.log(`Total new lines: ${newOrder.length}`);

    // Verify length hasn't changed (sanity check)
    if (lines.length !== newOrder.length) {
        console.warn(`Warning: Line count mismatch! Original: ${lines.length}, New: ${newOrder.length}`);
    }

    fs.writeFileSync(path, newOrder.join('\n'));
    console.log('Successfully reordered Layout.tsx');

} catch (err) {
    console.error(err);
}

File: reorder_menu.py

file_path = 'd:/SOFT-MEDIC/Antigravity/CURARE/frontend/src/components/Layout.tsx'

def reorder_lines():
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    print(f"Total original lines: {len(lines)}")

    # Define blocks by 0-indexed line ranges
    # ranges are [start_inclusive, end_exclusive]
    
    header = lines[0:109]
    inicio = lines[109:124]
    correos = lines[124:137]
    agenda = lines[137:154]
    usuarios = lines[154:171]
    doctores = lines[171:234]
    pacientes = lines[234:332]
    personal = lines[332:400]
    proveedores = lines[400:482]
    arancel = lines[482:497]
    egresos = lines[497:511]
    laboratorios = lines[511:623]
    gastos = lines[623:638]
    inventario = lines[638:654]
    hoja_diaria = lines[654:670]
    utilidades = lines[670:684]
    estadisticas = lines[684:787]
    configuracion = lines[787:918]
    footer = lines[918:]

    # Requested Order:
    # 1. Inicio
    # 2. Agenda
    # 3. Pacientes
    # 4. Doctores
    # 5. Laboratorios
    # 6. Proveedores
    # 7. Aranceles (Arancel)
    # 8. Personal
    # 9. Inventario
    # 10. Egresos Diarios
    # 11. Gastos Fijos
    # 12. Hoja Diaria
    # 13. Utilidades
    # 14. Estadisticas
    # 15. Correos
    # 16. Usuarios
    # 17. Configuración

    new_order = [
        *header,
        *inicio,
        *agenda,
        *pacientes,
        *doctores,
        *laboratorios,
        *proveedores,
        *arancel,
        *personal,
        *inventario,
        *egresos,
        *gastos,
        *hoja_diaria,
        *utilidades,
        *estadisticas,
        *correos,
        *usuarios,
        *configuracion,
        *footer
    ]

    print(f"Total new lines: {len(new_order)}")

    with open(file_path, 'w', encoding='utf-8') as f:
        f.writelines(new_order)

if __name__ == '__main__':
    reorder_lines()
